### DISCUSSION TITLE
MQTT subscribe to all topics

### DISCUSSION THREAD

**Discussion by eadmaster** [curl/curl#10807] [Q&A] [answered]
Is it possible to subscribe to **all** topics of an MQTT server like mosquitto supports?

`mosquitto_sub -v -t "#"`

[source](https://stackoverflow.com/questions/32904012/how-do-i-subscribe-to-all-topics-of-a-mqtt-broker)

Useful for debugging mqtt devices.

**Comment by bagder** [comment, marked as answer]
`mqtt://example.com/%23` ? `%23` being the `#` letter URL-encoded.

**Reply by eadmaster** [reply]
thanks, it seems to work, however i guess due to this i am getting no output on console

> [What does curl deliver as a response to a subscribe](https://everything.curl.dev/usingcurl/mqtt.html#what-does-curl-deliver-as-a-response-to-a-subscribe)
> It outputs two bytes topic length (MSB | LSB), the topic followed by the payload.

is there a way to get only the payload as printable strings?
i've tried with:
````
$ curl -s 'mqtt://localhost/%23' | strings
$ curl -s 'mqtt://localhost/%23' | tail -c +2
````
